dcos/dcos-ui
# DCOS-50840: Debug tab intermittently crashes on soak cluster

Handle the case of having an undefined
node id in the declined offers table and breaking the UI.

Closes https://jira.mesosphere.com/browse/DCOS-50840

## Testing
1. Go to services tab
2. Create a service with very high resource requirements
Example JSON:
```
{
  "id": "/new-service",
  "instances": 1,
  "portDefinitions": [],
  "container": {
    "type": "MESOS",
    "volumes": []
  },
  "cpus": 0.1,
  "mem": 1000000000000,
  "requirePorts": false,
  "networks": [],
  "healthChecks": [],
  "fetch": [],
  "constraints": [],
  "cmd": "sleep 1000000"
}
```
3. Run the service and go to it's debug tab
4. In `dcos-ui/plugins/services/src/js/components/DeclinedOffersTable.js` replace line 209
Currently:
```
          const node = MesosStateStore.getNodeFromHostname(hostname);
```
Replace with:
```
          const node = {id: null};
```
5. Verify that the UI doesn't break
6. Verify that the declined offers table at the bottom isn't shown
7. Test the same for pods

## Trade-offs
~1. The jira says that we should add tests for this scenario, but neither `DeclinedOffersTable`, nor `PodDebugContainer` and `ServiceDebugContainer` have defined test files and for me, creating complete new test files is out of scope for a 2-story point bug. I think we can create the tests as a follow-up.~
~2. Hiding the entire table because we can't render one column might seem over the top, but for me hiding just the Host column doesn't look good and isn't user friendly.~

## Dependencies
None.

## Screenshots
### Before
![Screenshot from 2019-04-03 15-26-13](https://user-images.githubusercontent.com/40791275/55478647-dd1d3100-5624-11e9-837b-7a049564a1f5.png)

### After
(no link)
![Screenshot from 2019-04-05 13-29-12](https://user-images.githubusercontent.com/40791275/55621742-09f95180-57a7-11e9-80a0-e903dde8bfe7.png)
